feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go

```diff
@@ -0,0 +1,469 @@
+			WithProgrammingResult(fluent.InstalledInRIB).
+			AsResult(),
+		chk.IgnoreOperationID(),
+	)
+
+	chk.HasResult(t, args.c.Results(t),
+		fluent.OperationResult().
+			WithIPv4Operation(ateIndirectNHCIDR).
+			WithOperationType(constants.Add).
+			WithProgrammingResult(fluent.InstalledInRIB).
+			AsResult(),
+		chk.IgnoreOperationID(),
+	)
+}
+
+// deleteRecursiveIPv4Entry verifies recursive IPv4 Entry for 198.51.100.0/24 -> 203.0.113.1/32 -> 192.0.2.6.
+// The entry for 203.0.113.1/32 is deleted. Verify that the traffic results in loss and removal from AFT.
+func deleteRecursiveIPv4Entry(t *testing.T, args *testArgs) {
+	args.c.Modify().DeleteEntry(t,
+		fluent.IPv4Entry().
+			WithPrefix(ateIndirectNHCIDR).
+			WithNetworkInstance(instance).
+			WithNextHopGroup(nhgIndex2))
+
+	if err := awaitTimeout(args.ctx, args.c, t, time.Minute); err != nil {
+		t.Fatalf("Could not program entries via c, got err: %v", err)
+	}
+
+	chk.HasResult(t, args.c.Results(t),
+		fluent.OperationResult().
+			WithIPv4Operation(ateIndirectNHCIDR).
+			WithOperationType(constants.Delete).
+			WithProgrammingResult(fluent.InstalledInRIB).
+			AsResult(),
+		chk.IgnoreOperationID(),
+	)
+}
+
+// testRecursiveIPv4Entry verifies recursive IPv4 Entry for 198.51.100.0/24 -> 203.0.113.1/32 -> 192.0.2.6.
+// The IPv4 Entry is verified through AFT Telemetry and Traffic.
+func testRecursiveIPv4Entry(t *testing.T, args *testArgs) {
+	setupRecursiveIPv4Entry(t, args)
+
+	aftsPath := args.dut.Telemetry().NetworkInstance(instance).Afts()
+	fptest.LogYgot(t, "AFTs", aftsPath, aftsPath.Get(t))
+
+	// Verify that the entry for 198.51.100.0/24 is installed through AFT Telemetry.
+	ipv4Entry := args.dut.Telemetry().NetworkInstance(instance).Afts().Ipv4Entry(ateDstNetCIDR).Get(t)
+	if got, want := ipv4Entry.GetPrefix(), ateDstNetCIDR; got != want {
+		t.Errorf("TestRecursiveIPv4Entry: ipv4-entry/state/prefix = %v, want %v", got, want)
+	}
+	if got, want := ipv4Entry.GetOriginProtocol(), telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_GRIBI; got != want {
+		t.Errorf("TestRecursiveIPv4Entry: ipv4-entry/state/origin-protocol = %v, want %v", got, want)
+	}
+	if got, want := ipv4Entry.GetNextHopGroupNetworkInstance(), instance; got != want {
+		t.Errorf("TestRecursiveIPv4Entry: ipv4-entry/state/next-hop-group-network-instance = %v, want %v", got, want)
+	}
+	nhgIndexInst := ipv4Entry.GetNextHopGroup()
+	if nhgIndexInst == 0 {
+		t.Errorf("TestRecursiveIPv4Entry: ipv4-entry/state/next-hop-group is not present")
+	}
+	nhg := args.dut.Telemetry().NetworkInstance(instance).Afts().NextHopGroup(nhgIndexInst).Get(t)
+	if got, want := nhg.GetProgrammedId(), uint64(nhgIndex); got != want {
+		t.Errorf("TestRecursiveIPv4Entry: next-hop-group/state/programmed-id = %v, want %v", got, want)
+	}
+
+	for nhIndexInst, nhgNH := range nhg.NextHop {
+		if got, want := nhgNH.GetIndex(), uint64(nhIndexInst); got != want {
+			t.Errorf("next-hop index is incorrect: want %v, got %v", want, got)
```

Comment:
nit: there are 4 instances of `want, got` here: messages should be using the order `got x, want y`.